Sources/SpeziBluetooth/CoreBluetooth/BluetoothPeripheral.swift

```diff
@@ -712,6 +734,12 @@ extension BluetoothPeripheral {
         }
     }
 
+    private func signalFullyDiscovered() {
+        _storage.signalFullyDiscovered()
+        connectContinuation?.resume()
+        connectContinuation = nil
```

Comment:
After resuming you need to make sure to also `signal()` the async semaphore `connectAccess` (that is protecting exclusive access to the `connectContinuation` property). Currently you are not doing that at all, meaning you can call `connect()` once and then it will block forever.